src/lcm/error_handling.py

```diff
@@ -42,12 +44,30 @@ def validate_value_function_array_integrity(
         states=state_action_space.states,
         period=period,
     )
-
+    extrapolation = False
+    if state_space_info.continuous_states is not None:
+        for key in state_space_info.continuous_states:
+            if (
+                invalid_states[key]
+                < state_space_info.continuous_states[key].to_jax()[0]
+            ).any() or (
+                invalid_states[key]
+                > state_space_info.continuous_states[key].to_jax()[-1]
+            ).any():
+                extrapolation = True
     log_dir = get_log_dir()
     file_path = log_dir / f"invalid_states_{_generate_unique_suffix()}.csv"
 
     invalid_states.to_csv(file_path, index=False, header=True)
-
+    if extrapolation:
+        raise InvalidValueFunctionError(
+            f"The value function array in period {period} contains NaN or infinite "
+            f"values. This might occur because the value function is extrapolated "
```

Comment:
I don't think we ever extrapolate the value function. 

In general, I am not sure I follow when this special case would happen (sorry it's been a while...). Could you maybe add a test for this particular error to show up?

Reply:
You are right, I will add test with an example of the error.

Reply:
The test I added should better illustrate the problem, during the solution of the model no `-inf` occurs in the value functions, but then during the simulation the agent is suddenly in a state where he has no valid actions. 

If you think the model is just too misspecified I'm fine with that, then I will just remove the error I added. I thought the behaviour was very odd, so I wanted to warn the user in cases where it can happen.